pkg/controller/import-controller.go

```diff
@@ -333,18 +127,62 @@ func (c *ImportController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 		lab = map[string]string{common.CDILabelKey: common.CDILabelValue}
 	}
 
-	pvc, err = updatePVC(c.clientset, pvc, anno, lab)
+	pvc, err = updatePVC(ic.clientset, pvc, anno, lab)
 	if err != nil {
 		return errors.WithMessage(err, "could not update pvc %q annotation and/or label")
 	}
 	return nil
 }
 
-// forget the passed-in key for this event and optionally log a message.
-func (c *ImportController) forgetKey(key interface{}, msg string) bool {
-	if len(msg) > 0 {
-		glog.V(3).Info(msg)
+//ProcessNextPvcItem ...
+func (ic *ImportController) ProcessNextPvcItem() bool {
+	key, shutdown := ic.queue.Get()
+	if shutdown {
+		return false
+	}
+	defer ic.queue.Done(key)
+
+	err := ic.syncPvc(key.(string))
+	if err != nil { // processPvcItem errors may not have been logged so log here
+		glog.Errorf("error processing pvc %q: %v", key, err)
+		return true
+	}
+	return ic.forgetKey(key, fmt.Sprintf("ProcessNextPvcItem: processing pvc %q completed", key))
+}
+
+func (ic *ImportController) syncPvc(key string) error {
+	pvc, exists, err := ic.pvcFromKey(key)
+	if err != nil {
+		return err
+	} else if !exists {
+		ic.podExpectations.DeleteExpectations(key)
+	}
+
+	if pvc == nil {
+		return nil
+	}
+	//check if AnnoEndPoint annotation exists
+	if !checkPVC(pvc, AnnEndpoint) {
+		return nil
+	}
+	glog.V(3).Infof("ProcessNextPvcItem: next pvc to process: %s\n", key)
+	return ic.processPvcItem(pvc)
+}
+
+//Run is being called from cdi-controller (cmd)
+func (ic *ImportController) Run(threadiness int, stopCh <-chan struct{}) error {
```

Comment:
Looks like all the controllers have basically the same Run() method.  Maybe it makes sense to add a helper function in base Controller that takes `ic.runPVCWorkers` as a parameter?  Actually, maybe a `runPVCWorkers` equivalent can be common as well.  `ic.ProcessNextPvcItem()` is the real controller dependent function

Reply:
:+1: Will move the run method. 
runPVCWorkers cannot get argument so I can't see how I go further...